src/abstractdataframe/io.jl

```diff
@@ -417,7 +423,9 @@ function printtable(io::IO,
             elseif isnothing(df[i, j])
                 print(io, nothingstring)
             else
-                if ! (etypes[j] <: Real)
+                if etypes[j] isa Markdown.MD
+                    show(io, "text/plain", df[i,j])
```

Comment:
why is this branch required here and not required in standard output to `stdout`? (I guess there is a reason but I am not sure what so please comment)

Reply:
It probably isn't necessary.

Reply:
Actually it looks like it matters, and `"text/plain"` should not be used I think, as it will mess up the output (I might be missing something here as I never used `Markdown` in my codes so please check):
```
julia> x = md"# **2** + *2* + ### 2"

julia> show(stdout, x) # it is just unformatted
# **2** + *2* + ### 2

julia> show(stdout, "text/plain", x) # you get a newline in the middle of the output here, which we probably do not want, but maybe it is OK and should be just properly wrapped
  2 + 2 + ### 2
  ≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡
```

Reply:
You are right that `show(stdout, "text/plain", x)` is a bad choice.  
However, there is another concern.  The `show(...)` methods for Markdown add a terminating `\n`.  This needs to be stripped.  Plus while it is possible to put a multi-line Markdown string in an HTML or LaTeX table, it doesn't work too well in a text table.

Use the DataFrames license as an example:
```julia
julia> md=Markdown.parse_file("C:\\Users\\nritchie\\.julia\\dev\\DataFrames\\LICENSE.md")
julia> df=DataFrame(a=[1,2,3],b=[3.0,4.0,5.0],c=[md"Link: [BBC](http://bbc.com)",md, md"Jump ![Julia Logo](https://docs.julialang.org/en/v1/assets/logo.svg) long"])
3×3 DataFrame
│ Row │ a     │ b       │ c                                                    │
│     │ Int64 │ Float64 │ Markdown.MD                                          │
├─────┼───────┼─────────┼──────────────────────────────────────────────────────┤
│ 1   │ 1     │ 3.0     │ [BBC](http://bbc.com)                                │
│ 2   │ 2     │ 4.0     │ DataFrames.jl is licensed under the MIT License:...  │
│ 3   │ 3     │ 5.0     │ ![Julia Logo](https://docs.julialang.org/en/v1/as... │
```
or as HTML
![image](https://user-images.githubusercontent.com/8323528/89241702-9c914700-d5cd-11ea-8b33-f5645d01b371.png)


To address both of these problems maybe we could use:
```julia 
function ourshow(io::IO, x::Markdown.MD)
    r=repr(x)
    f = findfirst(c->c=='\n', r)  # I don't think this will ever return nothing but...
    f = min(50, isnothing(f) ? length(f)+1 : f)
    return print(io, f<length(r) ? r[1:f-1]*"..." : r[1:f-1])
end
``` 
to "show.jl" and then using it in place of `show(io, "text/plain", x)`.  This truncates the output in a "text/plain" table but at a linebreak or in the case of a very long line at ?50? characters. 
 
However, this truncates the CSV/TSV output so we might modify
```julia
                if df[i,j] isa Markdown.MD                             # add
                    print(io, quotemark)                                  # add
                    escapedprint(io, repr(df[i,j]), quotestr)      # add
                    print(io, quotemark)                                  # add
                elseif ! (etypes[j] <: Real)
                    print(io, quotemark)
                    escapedprint(io, df[i, j], quotestr)
                    print(io, quotemark)
                else
                    print(io, df[i, j])
                end
```
to maintain the whole MD as text in the file.

Reply:
Looks good. We should to be in sync here with PrettyTables.jl so @ronisbr should comment here also.

Small comments:
* Instead of `...` better use `…` (`\ldots`)
* Instead of `isnothing(f) ? length(f)+1 : f` use `something`
* `r[1:f-1]` is incorrect, use `nextind` (otherwise you will get errors for non-ASCII indexing) and `SubString` (for speed)

To be clear what I want to say with the last point:
```
julia> s = "∀1"
"∀1"

julia> i = findfirst(==('1'), s)
4

julia> s[1:i-1]
ERROR: StringIndexError("∀1", 3)
```

Reply:
Yes, I am fortunate enough to rarely work with text manipulation so my UTF-8 skills are a little rusty. :smiley:

I'm going to continue to work on this as I have time today and over the next few days.  I'll create more test cases for some edge cases.  I may have identified an issue with the empty markdown string `md""`.   I'll make your suggested changes, fix the test case failures (which resulted from differences in Int on 32-bit and 64-bit systems.)  When  I'm happy with the changes, I'll push them so you can give them a look over.

Reply:
Excellent (it is a pleasure working with a contributor that "just knows" what should be done 😄).

Reply:
(actually looking at your code using `prevind` should be simpler than `nextind`)